### ISSUE TITLE
Versions in install_if_not_exists

### ISSUE THREAD

**Issue by containerman17** [ava-labs/hypersdk#1478]
If you check out the main branch at the latest commit (`7ce0081`):

```bash
git checkout main
cd examples/morpheusvm
./scripts/lint.sh
```

You'll see the following output:

```bash
Running 'golangci_lint gci' at: Fri Aug 30 10:49:17 AM UTC 2024
START: 'golangci_lint' at Fri Aug 30 10:49:17 AM UTC 2024
WARN [config_reader] The configuration option `run.skip-files` is deprecated, please use `issues.exclude-files`. 
WARN The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  
WARN The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused. 
ERRO [linters_context] interfacer: This linter is fully inactivated: it will not produce any reports. 
ERRO [linters_context] structcheck: This linter is fully inactivated: it will not produce any reports. 
cmd/morpheus-cli/cmd/chain.go:40:31: unused-parameter: parameter 'args' seems to be unused, consider removing or renaming it as _ (revive)
        RunE: func(_ *cobra.Command, args []string) error {
                                     ^
cmd/morpheus-cli/cmd/prometheus.go:20:31: unused-parameter: parameter 'args' seems to be unused, consider removing or renaming it as _ (revive)
        RunE: func(_ *cobra.Command, args []string) error {
                                     ^
FAIL: 'golangci_lint' failed at Fri Aug 30 10:49:18 AM UTC 2024
```

However, if you check the `morpheusvm-lint` CI action logs for the same commit [here](https://github.com/ava-labs/hypersdk/actions/runs/10630947896/job/29471013040), everything passes:

```
SUCCESS: 'gci' completed...
ALL SUCCESS!
```

This inconsistency suggests a possible issue with running the script from the wrong folder, a script mismatch, or a version mismatch.

**Comment by iFrostizz** [comment]
Possible leftover from https://github.com/ava-labs/hypersdk/pull/1443

**Comment by containerman17** [comment]
I cleaned up the system and started from a fresh `.devcontainer/default-go-env/devcontainer.json`. Below are the logs:

- **HyperSDK Lint Log**: [HyperSDK Lint Log](https://gist.github.com/containerman17/9bd252443a40060d3f1f307fc81e4e8d)
- **MorpheusVM Lint Log**: [MorpheusVM Lint Log](https://gist.github.com/containerman17/833dd03bae1078c0a1c48acdf2885518)

TLDR: the same

**Comment by containerman17** [comment]
vscode ➜ /workspaces/hypersdk/examples/morpheusvm (main) $ golangci-lint --version
golangci-lint has version 1.60.1 built with go1.23.0 from 3298c104 on 2024-08-14T01:15:05Z

Yes, you are right about the version

**Comment by containerman17** [comment]
The issue is that `install_if_not_exists` doesn't check the version, and Microsoft ships `golangci-lint` v1.16.1 by default. 

@iFrostizz, do you plan to upgrade the version in the next month or two? If not, I can add a post-install script to remove `golangci-lint`, or modify `install_if_not_exists` to check the version.

**Comment by containerman17** [comment]
We could simplify by replacing `install_if_not_exists` with `go install` every time. Running `go install` on an already installed package takes just 0.7 to 1 second, but it ensures the correct version is always installed.

**Comment by iFrostizz** [comment]
Good point, in this case if only 2 lints are left then maybe that could be best to replace `install_if_not_exists` by a hand-written installation of the specific version after having bumped the version ? It doesn't really seem a good idea to use different versions of the same binary across scripts

**Comment by containerman17** [comment]
Or even replace with `go run`, then no binary installation needed.

**Comment by aaronbuchwald** [comment]
The same is true for all of the places where we are using `install_if_not_exists` no?

**Comment by containerman17** [comment]
Yes, but everything else is working well so far. I’d prefer to keep the changes minimal here.
